packages/neptune-extension/src/common/utils/events.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  isFunction,
+  isPlainObject,
+} from 'lodash';
+
+export type EventHandler = (event: MouseEvent | object, params?: object) => void;
+
+interface HandlerObject {
+  handler: EventHandler
+  payload: any
+}
+
+function decomposeEventHandler(eventHandler?: EventHandler | HandlerObject) : [EventHandler?, HandlerObject?] {
+  if (isFunction(eventHandler)) {
+    return [eventHandler, undefined];
+  }
+
+  if (eventHandler && isPlainObject(eventHandler)) {
```

Comment:
If I didn't add `eventHandler &&` in if statement, TS would see handler in line 20 as HandlerObject | undefined.
Any how to improve this?